### ISSUE TITLE
Could I play a network video base h.264?

### ISSUE THREAD

**Issue by tshug** [mbebenita/Broadway#78]
I realtime get h.264 data from network by websocket,
can I play video by Broadway?
How to do ? give me a example?

**Comment by tshug** [comment]
I use ffmpeg get a usb camera h.264 data,
then post data to a java server.
use the java server send to chrome by websocket
I get binary stream (video data) in chrome
but I can not play the video data
How to do ?

**Comment by charliegreen** [comment]
I'm working on a similar problem, and I came across this. I hope you find it useful! https://github.com/mbebenita/Broadway/issues/9

**Comment by 131** [comment]
YES, broadway fits perfectly here, As you say, for live streaming, just connect to the remote source, split the incoming stream on each nal unit and send thoses units to Broadway via websockets.
I wrote a nodejs project that does just that and it's very efficient ! https://github.com/131/h264-live-player

**Comment by zettam** [comment]
I would love to have a solution to this as well. I'm currently sending raw camera data from raspberry pi and I would like to know if it's possible to render it through broadway

**Comment by mbebenita** [comment]
As long as you are using h.264 baseline profile things should work.

**Comment by zettam** [comment]
Do I need an mp4 wrapper ?
I'm directly sending everything from camera to broadway over sockets. Currently I see nothing but a black-ish square 200 by 200. I believe this is the default canvas.

**Comment by zettam** [comment]
Also how do I use the baseline profile?

**Comment by zettam** [comment]
```
var socket = io.connect('http://raspberry.pi.ip.address:port/');

var player = new Player();

document.body.appendChild(player.canvas);

socket.on('data', function (data) {
	console.log("x")
	player.decode(data);
});
```

I see x is being logged. When I console.log `data` it's many `ArrrayBuffer {}`s being logged

**Comment by soliton4** [comment]
no mp4 wrapper is needed
broadway is very good at decoding raw h264 and display it.
the mp4 player is a quick and dirty add on
so just leave the mp4 player aside and feed the stream directly to player.js

**Comment by zettam** [comment]
I was unable to make it work. Any example is appreciated.

**Comment by soliton4** [comment]
https://github.com/soliton4/BroadwayStream/blob/master/app/connected.pland#L28

**Comment by Burke111-DEV** [comment]
> ```
> var socket = io.connect('http://raspberry.pi.ip.address:port/');
> 
> var player = new Player();
> 
> document.body.appendChild(player.canvas);
> 
> socket.on('data', function (data) {
>     console.log("x")
>     player.decode(data);
> });
> ```
> 
> I see x is being logged. When I console.log `data` it's many `ArrrayBuffer {}`s being logged

I'm trying to do something like this too, but nothing happens.
The callback onPictureDecoded also never runs. Not sure what to look at next.

**Comment by soliton4** [comment]
this might very well be a problem with the profile. boradway only supports baseline profile. 
there are no good tools to investigate this in the wasm build, but if you use the c sources in the project and feed your stream to it you may see some error messages.
i hope this helps
